### DISCUSSION TITLE
GeoPandas 1.0.0 with GDAL and vsimem

### DISCUSSION THREAD

**Discussion by jan-dolezal** [geopandas/geopandas#3376] [Q&A] [answered]
I upgraded to geopandas 1.0.0 and everything was working fine.

The problem is, I was using a lot the `vsimem` file system.
I used to be able to directly write geodataframe to vsimem and call it with GDAL

Now I know, I can use io.ByteStream with gpd 1.0.0, but I have no idea how to give that to gdal. Can anyone point me in the direction?

My code used to be for example:

```
# before geopandas 1.0.0
df = gpd.read_file('local_file')
# process the dataframe and save it to vsimem
df.to_file('/vsimem/temp.gpkg')
gdal.Rasterize(out, '/vsimem/temp.gpkg')

#new code after 1.0.0 because /vsimem/ is not supported with pyogrio
with io.BytesIO() as stream:
    df = gpd.read_file(file)
    df.to_file(stream, driver="GPKG")


    # gdal ????

```

Now I can read the stream back to dataframe fine file is the same as while writing.
I could use `engine=fiona` but that would not be really helpful.

I know this might be question more towards the gdal, but lets see...

**Comment by jan-dolezal** [comment]
I think I have sorted this out, if someone can check, that this is reasonable, I would be very happy:

```
with io.BytesIO() as stream:
    df = gpd.read_file(file)
    df.to_file(stream, driver="GPKG")
    gdal.FileFromMemBuffer('/vsimem/somefile.gpkg', stream.read())
    gdal.Rasterize(out_file, '/vsimem/somefile.gpkg')

```

It is not ideal, but I can work with that.

**Comment by theroggy** [comment, marked as answer]
Sorry for the radio silence, but support for this has been added to the recently released [pyogrio 0.10](https://github.com/geopandas/pyogrio/releases/tag/v0.10.0).

**Reply by jan-dolezal** [reply]
Amazing, thank you
